x-pack/filebeat/input/http_endpoint/validate.go

```diff
@@ -52,3 +73,33 @@ func (v *apiValidator) ValidateHeader(r *http.Request) (int, error) {
 
 	return 0, nil
 }
+
+func (v *apiValidator) ValidateHmac(r *http.Request) (int, error) {
+	if v.hmacHeader != "" && v.hmacToken == "" {
+		return http.StatusMethodNotAllowed, fmt.Errorf("A hmacToken and has to be configured if hmacHeader is set")
+	}
+
+	if v.hmacToken != "" && v.hmacHeader != "" {
+		if len(r.Header.Get(v.hmacHeader)) == 0 {
+			return http.StatusInternalServerError, fmt.Errorf("The HMAC signature in the configured request header is empty")
+		}
+		s := hmac.New(sha1.New, []byte(v.hmacToken))
+		b, err := ioutil.ReadAll(r.Body)
```

Comment:
Not sure if reading this here (and not closing it) might cause issues to following attempts of reading it. I think you might need to read all, close the body, and create a new ReadCloser from the read bytes IIRC.

```
b, _ := ioutil.ReadAll(r.Body)
...
r.Body.Close()  //  must close
r.Body = ioutil.NopCloser(bytes.NewBuffer(b))
```